code/modules/mob/living/simple_animal/borer/borer.dm

```diff
@@ -0,0 +1,799 @@
+		return
+
+	var/list/choices = list()
+	for(var/mob/living/carbon/C in view(1,src))
+		if(C.stat == CONSCIOUS)
+			choices += C
+
+	if(world.time - used_dominate < dominate_cooldown)
+		src << "<span class='warning'>You cannot use that ability again so soon.</span>"
+		return
+
+	var/mob/living/carbon/M = input(src,"Who do you wish to dominate?") in null|choices
+
+
+	if(!M || !src) return
+	if(!Adjacent(M)) return
+
+	if(M.borer)
+		src << "<span class='warning'>You cannot paralyze someone who is already infested!</span>"
+		return
+
+	src.layer = MOB_LAYER
+
+	src << "<span class='warning'>You focus your psychic lance on [M] and freeze their limbs with a wave of terrible dread.</span>"
+	M << "<span class='userdanger'>You feel a creeping, horrible sense of dread come over you, freezing your limbs and setting your heart racing.</span>"
+	M.Stun(4)
+
+	used_dominate = world.time
+
+/mob/living/simple_animal/borer/verb/release_victim()
+	set category = "Borer"
+	set name = "Release Host"
+	set desc = "Slither out of your host."
+
+	if(!victim)
+		src << "<span class='userdanger'>You are not inside a host body.</span>"
+		return
+
+	if(stat != CONSCIOUS)
+		src << "<span class='userdanger'>You cannot leave your host in your current state.</span>"
+
+	if(!victim || !src) return
+
+	if(leaving)
+		leaving = 0
+		src << "<span class='userdanger'>You decide against leaving your host.</span>"
+		return
+
+	src << "<span class='userdanger'>You begin disconnecting from [victim]'s synapses and prodding at their internal ear canal.</span>"
+
+	if(victim.stat != DEAD)
+		host << "<span class='userdanger'>An odd, uncomfortable pressure begins to build inside your skull, behind your ear...</span>"
+
+	leaving = 1
+
+	spawn(100)
+
+		if(!victim || !src) return
+		if(!leaving) return
+		if(controlling) return
+
+		if(src.stat != CONSCIOUS)
+			src << "<span class='userdanger'>You cannot release your host in your current state.</span>"
+			return
+
+		src << "<span class='userdanger'>You wiggle out of [victim]'s ear and plop to the ground.</span>"
+		if(victim.mind)
+			host << "<span class='danger'>Something slimy wiggles out of your ear and plops to the ground!</span>"
+			host << "<span class='danger'>As though waking from a dream, you shake off the insidious mind control of the brain worm. Your thoughts are your own again.</span>"
+
+		leave_victim()
+
+
+/mob/living/simple_animal/borer/verb/jumpstart()
+	set category = "Borer"
+	set name = "Jumpstart Host"
+	set desc = "Brings your host back from the dead."
+
+	if(!victim)
+		src << "<span class='warning'>You need a host to be able to use this.</span>"
+		return
+
+	if(docile)
+		src << "<span class='warning'>You are feeling too docile to use this!</span>"
+		return
+
+	if(chemicals < 250)
+		src << "<span class='warning'>You need 250 chemicals to use this!</span>"
+		return
+
+	if(victim.stat == DEAD)
+		dead_mob_list -= victim
+		living_mob_list += victim
+		victim.tod = null
+		victim.setToxLoss(0)
```

Comment:
do we not have a preexisting proc to heal mobs?
might be better to use if we do, @tgstation/commit-access anyone know?